stdlib/src/utils/static_tuple.mojo

```diff
@@ -254,3 +314,139 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         var base_ptr = Pointer[Self.type].address_of(self.array).address
         var ptr = __mlir_op.`pop.array.gep`(base_ptr, Int(0).value)
         return Pointer(ptr)
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct Array[ElementType: CollectionElement, size: Int](Sized):
```

Comment:
**Suggestion** I'm good with having `Array` be defined here rather than in `collections/array.mojo` for now to make the migration easier for us internally (and we can do the move once we remove the `StaticTuple`).

Reply:
Given we have a List now (following Python convention) I think a bunch of folk will be confused about Array (the name in a lot of other languages) vs List.  This is a C-style inline array, so WDYT about a name like "InlineArray" to make this more explicit?

Reply:
@lattner Shouldn't the emphasis more on `Static` (having a size known at compile time) than `Inline`?

There is one advantage in keeping `Array`: it feels nature when/if we have `NDArray`.

Reply:
-1 for "Array" as array is already defined in the python stdlib: https://docs.python.org/3/library/array.html (note that this module doesn't make much sense in mojo as python's array's use case is covered by Mojo's List). I can already see the confused faces of Python devs when I'll try to explain them that Mojo's array is different from python's array 😁 

I'm fine with whatever other name, StaticArray, InlineArray, CArray are all fine by me...